src/mpsc/errors.rs

```diff
@@ -1,6 +1,26 @@
 //! Errors returned by channels.
 use core::fmt;
 
+/// Error returned by the [`Sender::send_timeout`] or [`Sender::send_ref_timeout`] 
+/// (and [`StaticSender::send_timeout`]/[`StaticSender::send_ref_timeout`]) methods.
+///
+/// [`Sender::send_timeout`]: super::Sender::send_timeout
+/// [`Sender::send_ref_timeout`]: super::Sender::send_ref_timeout
+/// [`StaticSender::send_timeout`]: super::StaticSender::send_timeout
+/// [`StaticSender::send_ref_timeout`]: super::StaticSender::send_ref_timeout
+#[non_exhaustive]
```

Comment:
since this error type is only returned by the blocking channel, we can make it `#[cfg(feature = "std")]`
```suggestion
#[cfg(feature = "std")]
#[non_exhaustive]
```